### PR TITLE
revise comments and harden nx_mount logic

### PR DESCRIPTION
**Pull request by yf13**
## Summary

This patch contains  improvements to harden logic of nx_mount():
- currently some variables may not get populated by called functions, thus initialization is added.
- empty string device path is possible when multiple type of file systems are enabled.

It also revised comments for mm_heap/mm.h

## Impact

file system mount operations

## Testing

Checked with CanMV230

### CODE REVIEW

File: fs/mount/fs_mount.c [hunk 1]
```diff
@@ -293,7 +293,7 @@ int nx_mount(FAR const char *source, FAR const char *target,
 
   /* Find the specified filesystem. Try the block driver filesystems first */
 
-  if (source != NULL &&
+  if (source != NULL && strlen(source) > 0 &&
```
Comment (xiaoxiang781216) [inline fs/mount/fs_mount.c:296]: why need strlen

File: fs/mount/fs_mount.c [hunk 2]
```diff
@@ -310,7 +310,7 @@ int nx_mount(FAR const char *source, FAR const char *target,
           goto errout_with_inode;
         }
     }
-  else if (source != NULL &&
+  else if (source != NULL && strlen(source) > 0 &&
```
Comment (xiaoxiang781216) [inline fs/mount/fs_mount.c:313]: ditto

File: fs/mount/fs_mount.c (hunk 1 above)
Comment (yf13) [reply fs/mount/fs_mount.c:296]: When source is "" and without this, there are duplicated "mm_free: Freeing ..." log during my test.

**Review by xiaoxiang781216 — APPROVED** [review]

File: fs/mount/fs_mount.c (hunk 1 above)
Comment (xiaoxiang781216) [reply fs/mount/fs_mount.c]: It's better than strlen

File: mm/mm_heap/mm.h [hunk 3]
```diff
@@ -121,8 +121,10 @@
 #define MM_ALIGN_UP(a)   (((a) + MM_GRAN_MASK) & ~MM_GRAN_MASK)
 #define MM_ALIGN_DOWN(a) ((a) & ~MM_GRAN_MASK)
 
-/* An allocated chunk is distinguished from a free chunk by bit 0
- * of the 'preceding' chunk size.  If set, then this is an allocated chunk.
+/* Due to alignments the lowest two bits of valid chunk size are always
```
Comment (davids5) [inline mm/mm_heap/mm.h:124]:
```suggestion
/* Due to alignment, the lowest two bits of valid chunk size are always
```

File: mm/mm_heap/mm.h [hunk 4]
```diff
@@ -121,8 +121,10 @@
...
+ * zero, thus the two bits are reused to depict allocation status: the bit
```
Comment (davids5) [inline mm/mm_heap/mm.h:125]:
```suggestion
 * zero, thus the two bits are reused to depict allocation status: bit
```

**Review by davids5 — CHANGES_REQUESTED** [review]
See inline

**Comment by beginner77-git** [comment]
> ## Summary
> This patch contains improvements to harden logic of nx_mount():
> 
> * currently some variables may not get populated by called functions, thus initialization is added.
> * empty string device path is possible when multiple type of file systems are enabled.
> 
> It also revised comments for mm_heap/mm.h
> 
> ## Impact
> file system mount operations
> 
> ## Testing
> Checked with CanMV230

**Pull request merged**